packages/x-components/src/x-modules/search-box/components/search-input-placeholder.vue

```diff
@@ -0,0 +1,179 @@
+      <span
+        v-if="animatedPlaceholder && isAnimationVisible"
+        :key="animatedPlaceholder"
+        class="x-search-input-placeholder__placeholder"
+        data-test="animated-placeholder"
+      >
+        {{ animatedPlaceholder }}
+      </span>
+      <span
+        v-else-if="$slots.default"
+        class="x-search-input-placeholder__default-placeholder"
+        data-test="default-placeholder"
+      >
+        <!-- @slot Default static placeholder. -->
+        <slot />
+      </span>
+    </component>
+  </div>
+</template>
+
+<script lang="ts">
+  import Vue from 'vue';
+  import { Component, Prop } from 'vue-property-decorator';
+  import { State } from '../../../components/decorators/store.decorators';
+  import { XOn } from '../../../components/decorators/bus.decorators';
+  import { xComponentMixin } from '../../../components/x-component.mixin';
+  import { searchBoxXModule } from '../x-module';
+
+  /**
+   * Renders a placeholder that is displayed on top of the search input and
+   * can be animated when given an array of animated placeholders.
+   *
+   * @public
+   */
+  @Component({
+    mixins: [xComponentMixin(searchBoxXModule)]
+  })
+  export default class SearchInputPlaceholder extends Vue {
+    @State('searchBox', 'query')
+    public query!: string;
+
+    protected get isQueryEmpty(): boolean {
+      return this.query.length === 0;
+    }
+
+    /**
+     * List of placeholders that will be animated.
+     *
+     * @public
+     */
+    @Prop({ default: () => [] })
+    protected animatedPlaceholders!: string[];
+
+    /**
+     * Animation component that will be used to animate the placeholders.
+     *
+     * @remarks This component must emit an `after-enter` event when the
+     * enter animation has finished and that's when the next placeholder will
+     * be shown.
+     *
+     * @public
+     */
+    @Prop({ default: 'div' })
+    protected animation!: Vue | string;
+
+    protected currentPlaceholder = -1;
+    protected isAnimationVisible = false;
+
+    /**
+     * A computed property that returns the current animated placeholder.
+     *
+     * @returns String with the placeholder text.
+     *
+     * @internal
+     */
+    protected get animatedPlaceholder(): string | undefined {
+      if (this.$props.animatedPlaceholders.length === 0) {
+        return undefined;
+      }
+      return this.$props.animatedPlaceholders[this.currentPlaceholder];
+    }
+
+    /**
+     * Loops through the animated placeholders;.
+     */
+    protected showNextPlaceholder(): void {
+      if (this.animatedPlaceholders.length > 0) {
+        this.currentPlaceholder = (this.currentPlaceholder + 1) % this.animatedPlaceholders.length;
+      }
+    }
+
+    /**
+     * When event {@link SearchBoxXEvents.UserFocusedSearchBox} is
+     * emitted, the next animated placeholder is displayed.
+     *
+     * @internal
+     */
+    @XOn(['UserFocusedSearchBox'])
```

Comment:
According to the feature description, the animation should start on hover and disappear when the input is focused instead of starting on focus (`UserFocusedSearchBox`):

> Hover scenario
Given a shopper who is going to type in the search box
when the user hovers the search box
then the placeholder is animated showing different messages
when the user clicks on the search box and the input get focused
then the placeholder disappears to start typing.

> Normal scenario
Given a shopper who is going to type in the search box
then the placeholder is animated showing different messages
when the user clicks on the search box and the input get focused
then the placeholder disappears to start typing.

With the current implementation, if the `SearchInput` component has the `autofocus` prop disabled, we won't see the animation unless the input is focused. 

But, in my opinion, I would start the animation as soon as the input is rendered and finish it when the user has typed something in the search input. Maybe in another iteration, we could define a finite number of times the animation will loop, if desired. 

What do you think about it @alonsogarciapablo @herrardo?